File: plugin/zeppelin/src/main/scala/jetbrains/zeppelin/idea/settings/notebook/NotebookBrowserDialog.scala
package jetbrains.zeppelin.idea.settings.notebook

import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.DialogWrapper
import javax.swing.{JComponent, SwingConstants}
import jetbrains.zeppelin.components.ZeppelinComponent
import jetbrains.zeppelin.models.Notebook

import scala.collection.JavaConverters._

class NotebookBrowserDialog(project: Project, defaultNotebook: Notebook) extends DialogWrapper(project) {
  val title = s"Notebook browser"
  private val myPanel = new NotebookBrowserForm()
  private val actionService = ZeppelinComponent.connectionFor(project).service

  setTitle(title)
  setButtonsAlignment(SwingConstants.CENTER)
  init()

  override def createCenterPanel(): JComponent = myPanel.getContentPane

  override def doOKAction(): Unit = {
    val currentNotebooks = myPanel.getNotebooks.asScala.toList
    val notebooks = actionService.updateNotebooksTo(currentNotebooks)
    myPanel.setNotebooks(notebooks.asJava)
    super.doOKAction()
  }

  override def init(): Unit = {
    super.init()
    initDataModel()
  }

  def initDataModel(): Unit = {
    val notebooks = getOriginalNotebooks
    myPanel.initDataModel(notebooks.asJava)
  }

  /**
    * Open from and get selected result
    *
    * @return tuple (is value selected, selected value)
    */
  def openAndGetResult(): (Boolean, Option[Notebook]) = {
    if (!showAndGet()) return (false, None)

    val value = myPanel.getSelectedValue
    if (value != null &&
      value.name == ZeppelinComponent.connectionFor(project).getZeppelinSettings.defaultNotebookName) {
      return (true, None)
    }
    (true, Option(value))

  }

  private def getOriginalNotebooks = actionService.getNotebooksList
}

File: zeppelin/implementation/src/org/intellij/plugin/zeppelin/idea/settings/notebook/NotebookBrowserDialog.kt
package org.intellij.plugin.zeppelin.idea.settings.notebook

import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.DialogWrapper
import org.intellij.plugin.zeppelin.components.ZeppelinComponent
import org.intellij.plugin.zeppelin.models.Notebook
import javax.swing.JComponent
import javax.swing.SwingConstants

class NotebookBrowserDialog(val project: Project) : DialogWrapper(project) {
    private val myPanel = NotebookBrowserForm()
    private val actionService = ZeppelinComponent.connectionFor(project).service

    init {
        title = "Notebook browser"
        setButtonsAlignment(SwingConstants.CENTER)
        init()
    }

    override fun createCenterPanel(): JComponent = myPanel.contentPane

    override fun doOKAction() {
        val currentNotebooks = myPanel.notebooks
        val notebooks = actionService.updateNotebooksTo(currentNotebooks).sortedBy { it.name }
        myPanel.notebooks = notebooks
        super.doOKAction()
    }

    override fun init() {
        super.init()
        initDataModel()
    }

    fun initDataModel() {
        val notebooks = getOriginalNotebooks().sortedBy { it.name }
        myPanel.initDataModel(notebooks)
    }

    /**
     * Open from and get selected result
     *
     * @return tuple (is value selected, selected value)
     */
    fun openAndGetResult(): Pair<Boolean, Notebook?> {
        if (!showAndGet()) return Pair(false, null)

        val value = myPanel.selectedValue ?: return Pair(false, null)
        if (value.name == ZeppelinComponent.connectionFor(project).getZeppelinSettings().defaultNotebookName) {
            return Pair(true, null)
        }
        return Pair(true, value)
    }

    private fun getOriginalNotebooks() = actionService.getNotebooksList()
}